### DISCUSSION TITLE
Global with CI

### DISCUSSION THREAD

**Discussion by micheelengronne** [npm/feedback#161] [CLI] [answered]
Is it possible to have a compatibility between the `global` flag and the `ci` flag ?

I woulk like to have a perfectly immutable system when using npm and it is not quite the case when installing global packages.

I install global package using ansible but basically it is `npm install -g` in a directory where a `package.json` file and a `package-lock.json` file are placed.

As `npm install -g` can update the lockfile I don't have any certainty that 2 deploys of the same `package.json` globally will give the same code.

That breaks immutability, a fundamental concept of devops and an absolute need to make complex systems stable.

**Comment by MylesBorins** [comment, marked as answer]
hey @micheelengronne,

have you tried running `npm i -g --no-save`?

**Reply by micheelengronne** [reply]
In fact, I used another way. I installed my packages locally with `ci` and I `link` them to global. The only problem is that binaries are not symlinked in the global bin directory. So we have to put the bin `PATH` in `ENV`.
